AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+    )
+}
+
+/**
+ * Shows a row for one topic.
+ *
+ * @param topic The topic title.
+ * @param expanded Whether the row should be shown expanded with the topic body.
+ * @param onClick Called when the row is clicked.
+ */
+@Composable
+private fun TopicRow(topic: String, expanded: Boolean, onClick: () -> Unit) {
+    TopicRowSpacer(visible = expanded)
+    Surface(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(onClick = onClick),
+        elevation = 2.dp
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(16.dp)
+                .animateContentSize()
+        ) {
+            Row {
+                Icon(
+                    imageVector = Icons.Default.Info,
+                    contentDescription = stringResource(R.string.info)
+                )
+                Spacer(modifier = Modifier.width(16.dp))
+                Text(
+                    text = topic,
+                    style = MaterialTheme.typography.body1
+                )
+            }
+            if (expanded) {
+                Spacer(modifier = Modifier.height(8.dp))
+                Text(
+                    text = stringResource(R.string.lorem_ipsum),
+                    textAlign = TextAlign.Justify
+                )
+            }
+        }
+    }
+    TopicRowSpacer(visible = expanded)
+}
+
+/**
+ * Shows a separator for topics.
+ */
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun TopicRowSpacer(visible: Boolean) {
+    AnimatedVisibility(visible = visible) {
+        Spacer(modifier = Modifier.height(8.dp))
+    }
+}
+
+/**
+ * Shows the bar that holds 2 tabs.
+ *
+ * @param backgroundColor The background color for the bar.
+ * @param onTabSelected Called when the tab is switched.
+ */
+@Composable
+private fun HomeTabBar(
+    backgroundColor: Color,
+    currentTab: Int,
+    onTabSelected: (index: Int) -> Unit
+) {
+    TabRow(
+        selectedTabIndex = currentTab,
+        backgroundColor = backgroundColor,
+        indicator = { tabPositions ->
+            HomeTabIndicator(tabPositions, currentTab)
+        }
+    ) {
+        HomeTab(
+            icon = Icons.Default.Home,
+            contentDescription = stringResource(R.string.home),
+            title = stringResource(R.string.home),
+            onClick = { onTabSelected(0) }
+        )
+        HomeTab(
+            icon = Icons.Default.AccountBox,
+            contentDescription = stringResource(R.string.work),
+            title = stringResource(R.string.work),
+            onClick = { onTabSelected(1) }
+        )
+    }
+}
+
+/**
+ * Shows an indicator for the tab.
+ *
+ * @param tabPositions The list of [TabPosition]s from a [TabRow].
+ * @param selectedTabIndex The index of the currently selected tab.
+ */
+@Composable
+private fun HomeTabIndicator(
+    tabPositions: List<TabPosition>,
+    selectedTabIndex: Int
+) {
+    val transition = updateTransition(selectedTabIndex)
```

Comment:
What about extracting the transition to it's own fun to separate animating values from the content?

Reply:
That's a useful technique, but there's no problem in putting transiiton in the same composable as the UI. The UI code here is very short so I think it is good.